> LINK : fatal error LNK1181: cannot open input file 'stdio_msvc12_64.obj'

The druntime lib dir isn't in the libpath, you need to copy the objects to phobos or change the libpath.

--- a/src/libmscoff.d
+++ b/src/libmscoff.d
@@ -151,7 +151,7 @@ public:
                 offset += MSCoffLibHeader.sizeof;
                 char* endptr = null;
                 uint size = strtoul(cast(char*)header.file_size, &endptr, 10);
-                if (endptr >= &header.file_size[10] || *endptr != ' ')
+                if (endptr >= header.file_size.ptr + 10 || *endptr != ' ')
                 {
                     reason = __LINE__;
                     goto Lcorrupt;

--- a/src/link.d
+++ b/src/link.d
@@ -248,7 +248,7 @@ extern (C++) int runLINK()
             {
                 lflags = getenv("LFLAGS_VS14");
                 if (!lflags)
-                    lflags = global.params.is64bit ? "stdio_msvc14_64.obj" : "stdio_msvc14_32mscoff.obj";
+                    lflags = "legacy_stdio_definitions.lib";
                 // environment variables UniversalCRTSdkDir and UCRTVersion set
                 // when running vcvarsall.bat x64
                 if (const(char)* UniversalCRTSdkDir = getenv("UniversalCRTSdkDir"))
@@ -267,11 +267,12 @@ extern (C++) int runLINK()
             else
             {
                 lflags = getenv("LFLAGS_VS12");
-                if (!lflags)
-                    lflags = global.params.is64bit ? "stdio_msvc12_64.obj" : "stdio_msvc12_32mscoff.obj";
             }
-            cmdbuf.writeByte(' ');
-            cmdbuf.writestring(lflags);
+            if (lflags)
+            {
+                cmdbuf.writeByte(' ');
+                cmdbuf.writestring(lflags);
+            }
             char* p = cmdbuf.peekString();
             const(char)* lnkfilename = null;
             size_t plen = strlen(p);

